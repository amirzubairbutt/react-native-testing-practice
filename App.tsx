import { View, Text } from 'react-native'
import React from 'react'
import sum from './src/example/sum'

const App = () => {
  return (
    <View>
      <Text>App</Text>
      <Text>{sum(5, 5)}</Text>
    </View>
  )
}

export default App