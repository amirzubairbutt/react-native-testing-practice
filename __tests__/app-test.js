import React from "react";
import renderer from 'react-test-renderer'
import App from "../App";
import { expect, jest } from "@jest/globals";

test('render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
});

// jest.mock('Test', () => {
//     const RealCopmonent = jest.requireActual('Test');
//     const React = require('react');
//     class Text extends React.Component {
//         render() {
//             return React.createElement('Text', this.props, this.props.childern);
//         }
//     }
//     Text.propTypes = RealCopmonent.propTypes;
//     return Text
// })