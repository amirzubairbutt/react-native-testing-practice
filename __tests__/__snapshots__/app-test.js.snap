// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View>
  <Text>
    App
  </Text>
  <Text>
    10
  </Text>
</View>
`;
